nanoFramework.Device.Bluetooth/BluetoothLEDevice.cs

```diff
@@ -315,36 +324,67 @@ internal GattReadResult ReadAttributeValue(ushort AttributeHandle)
             GattCommunicationStatus status = GattCommunicationStatus.Unreachable;
             Buffer buffer = null;
             ushort result = 0;
+            bool securityStarted = false;
+            bool retryRead;
 
             lock (lockObj)
             {
-                _completedEvent.Reset();
-
-                //Debug.WriteLine($"# ReadValue start con:{ConnectionHandle} attrHandle:{AttributeHandle}  ");
-                result = NativeStartReadValue(ConnectionHandle, AttributeHandle);
-                if (result == 0)
+                do
                 {
-                    // Read started, wait for completed event
-                    if (_completedEvent.WaitOne(operationTimeout, false))
+                    retryRead = false;
+
+                    //Debug.WriteLine($"# ReadAttributeValue");
+
+                    _completedEvent.Reset();
+
+                    //Debug.WriteLine($"# ReadValue start con:{ConnectionHandle} attrHandle:{AttributeHandle}  ");
+                    result = NativeStartReadValue(ConnectionHandle, AttributeHandle);
+                    if (result == 0)
```

Comment:
won't it be easier with an enum? Especially looking further with the code, there are cases, so it will be easier to maintain with an enum. And they don't cost more than normal int.

Reply:
Probably, will look at doing that